### PR TITLE
doc: Update architectures documentation (from Incus)

### PR DESCRIPTION
**Pull request by kadinsayani**
cherry-picked from: lxc/incus#354

### CODE REVIEW

**Comment by tomponline** [comment]
@kadinsayani added " (from Incus)" to PR description to help with attribution

File: doc/.wordlist.txt [hunk 1]
```diff
@@ -25,6 +25,15 @@ Kubeflow
...
 linter
+kB
+kbit
+KiB
+kibi
+Kibit
+KVM
+lookups
+Loongarch
+LRU
```
Comment (ru-fu) [inline doc/.wordlist.txt:36]: Most of these terms aren't needed. Unless I missed any, the only one used in this PR is "Loongarch".
Also, we don't edit `.wordlist.txt` but instead add exceptions to `.custom_wordlist.txt` (see the comment at the top of the file).

File: doc/architectures.md [hunk 2]
```diff
@@ -22,11 +22,25 @@ ID    | Name          | Notes                           | Personalities
...
 LXD cares only about the kernel architecture, not the particular userspace flavor as determined by the toolchain.
 
-That means that LXD considers ARMv7 hard-float to be the same as ARMv7 soft-float and refers to both as `armv7`.
+That means that LXD considers ARMv7 hard-float to be the same as ARMv7 soft-float and refers to both as `armv7l`.
 If useful to the user, the exact userspace ABI may be set as an image and container property, allowing easy query.
 ```
+
+## Virtual-machine support
+
+LXD only supports running virtual-machines on the following host architectures:
```
Comment (ru-fu) [inline doc/architectures.md:38]:
```suggestion
## Virtual machine support

LXD only supports running virtual machines on the following host architectures:
```

**Review by ru-fu — COMMENTED** [review]
The changes should probably be separate commits on top of the cherry-picks?

**Comment by ru-fu** [comment]
You'll need "Loongarch" in the `.custom_wordlist.txt` file after all. :)

**Comment by kadinsayani** [comment]
"Loongarch" is currently in the ```.custom_wordlist.txt``` file. To clarify, does it also need to be included in ```.wordlist.txt```?

**Comment by ru-fu** [comment]
> "Loongarch" is currently in the `.custom_wordlist.txt` file. To clarify, does it also need to be included in `.wordlist.txt`?

No, only in `.custom_wordlist.txt`. I didn't check - I just saw that the documentation check is read.

`.custom_wordlist.txt` actually has `LoongArch` (different capitalisation), and Google confirms that is actually correct. So it should be changed for the cherry pick.

**Review by ru-fu — APPROVED** [review]
Thanks! :)

**Pull request merged**